lib/column/tuple.go

```diff
@@ -196,8 +196,10 @@ func setJSONFieldValue(field reflect.Value, value reflect.Value) error {
 
 	// check if our target is a string
 	if field.Kind() == reflect.String {
-		field.Set(reflect.ValueOf(fmt.Sprint(value.Interface())))
-		return nil
+		if v := reflect.ValueOf(fmt.Sprint(value.Interface())); v.Type().AssignableTo(field.Type()) {
+			field.Set(v)
+			return nil
+		}
```

Comment:
Super important:
see test `TestCustomArray`
in case we have custom type instead of string -> field.Set(v) , where field is a custom string type and v is string type - reflect package will panic
So to fix this possible issue firstly we're checking if value can be assignable to field (if not - just go to next condition and trying to convert val to field).

Reply:
Will prepare go/play in a moment to show this issue

Reply:
Additional explanation: https://go.dev/play/p/UrtEjsnZEG0